clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobWorkerImpl.java

```diff
@@ -234,24 +239,89 @@ private void getPollInterval() {
   }
 
   private void handleJob(final ActivatedJob job) {
+    final boolean notEnqueued = !tryToEnqueueJob(job);
+    if (notEnqueued) {
+      // TODO logic to handle the faults?
+      // The job is likely to timed out anyway, so we potentially just skip it and don't need to do
+      // any further error handling
+
+      // Interesting is here whether we maybe want to explode, since adding other jobs from the
+      // batch also doesn't make much sense.
+      return;
```

Comment:
❓ Something I would like to discuss with you @npepinpe

Reply:
Don't we want to block forever? :thinking: Or at least until we're interrupted and/or closed.

Reply:
Hm, unsure about polling. In streaming I expect we just want to block the `onNext` right?

Reply:
We discussed this and agreed on to keep this as an improvement, but we should also be able to disable this so users can get there old worker behavior back.